web/src/components/Navigation/TopNav.tsx

```diff
@@ -14,22 +15,39 @@ import IconButton from "../Inputs/IconButton";
 import LinkIconButton from "../Inputs/LinkIconButton";
 import SearchBar from "../Inputs/SearchBar";
 
-export default function TopNav() {
+interface Props {
+  setSearchValue: (value: string) => void;
+}
+
+export default function TopNav({ setSearchValue }: Props) {
   //#region Hooks
 
   const router = useRouter();
 
   const { scrollY } = useScroll();
 
   const [navHidden, setNavHidden] = useState(false);
-  const [searchTerm, setSearchTerm] = useState("");
+  const [localSearchValue, setLocalSearchValue] = useState("");
+
+  // Debounce the search value (don't immediately update it after every keystroke)
+  const debouncedSetSearchValue = useMemo(
+    () => debounce(setSearchValue, 300),
+    []
+  );
```

Comment:
main thing would be having the input be controlled and not depend on the debounced input. We could technically move this debouncing logic to the component that actually fetches the posts based on the searchValue

Reply:
I do think it would be better to move the debouncing logic outside of this component.

Even though this is a search bar, that doesn't necessarily imply that a search involves network calls. By putting the debouncing in here, the component doesn't work as well in a case where we would want to search a local cache or other synchronously accessible set of data. That being said, we don't necessarily need that right now, but that's why I don't prefer this.

Reply:
If you just want this merged though, we can always change that later if we need to and I know we're up against a deadline today